quiz/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"bufio"
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+)
+
+type Quiz struct {
+	filename       string
+	timeLimit      int
+	shuffle        bool
+	problems       [][]string
+	totalScore     int
+	totalQuestions int
+}
+
+func (q *Quiz) readProblems() {
+	csvfile, err := os.Open(q.filename)
+	if err != nil {
+		log.Fatalln("Couldn't open csv file ", err)
+	}
+	r := csv.NewReader(csvfile)
+	records, err := r.ReadAll()
+	if err != nil {
+		log.Fatalln(err)
+	}
+	q.problems = records
+}
+func (q *Quiz) shuffleProblems() {
+	rand.Seed(time.Now().UnixNano())
+	rand.Shuffle(len(q.problems), func(i, j int) { q.problems[i], q.problems[j] = q.problems[j], q.problems[i] })
+}
+
+func (q *Quiz) printProblems() {
+	in := bufio.NewReader(os.Stdin)
+	for i := 0; i < len(q.problems); i++ {
+		question, answer := q.problems[i][0], q.problems[i][1]
+		fmt.Println("What is " + question + " ?")
+		userAnswer, _ := in.ReadString('\n')
```

Comment:
In Go, you'll typically see people iterating through slices using the range clause.
i.e., your code might look something like:
```suggestion
	for _, problem := range q.problems {
		question, answer := problem[0], problem[1]
		fmt.Printf("What is %s?\n", question)
		userAnswer, _ := in.ReadString('\n')
		
		// Rest of your code here
	}
```
Also note the use of fmt.Printf() instead of fmt.Println(). It's not actually important in this case, given that question is just a string, but I thought I'd make a note just in case you weren't familiar.

Reply:
Oh okay. I'll make sure that I follow this in the future.